api/types/resource.go

```diff
@@ -468,15 +468,10 @@ func MatchLabels(resource ResourceWithLabels, labels map[string]string) bool {
 	return true
 }
 
-// LabelPattern is a regexp that describes a valid label key
-const LabelPattern = `^[a-zA-Z/.0-9_:*-]+$`
-
-var validLabelKey = regexp.MustCompile(LabelPattern)
-
 // IsValidLabelKey checks if the supplied string matches the
 // label key regexp.
 func IsValidLabelKey(s string) bool {
```

Comment:
Why do we need two names for the same func?

Reply:
This was done to avoid an import cycle in the `header` package and still keep the logic identical between the two.

Reply:
Feels like we're covering up a code smell here, but I'll leave it up to you.

Reply:
Yeah, removing something entirely from an API module that we intend for others to use has to be done with great care, which is why I'm being a bit picky on reviews for what seems like simple things under `api/`. Once it's there, people will come to rely on it, and we don't want to break them, which mean we should be confident in and happy with the APIs we export.